scripts/utils/generate_ts.sh

```diff
@@ -21,82 +21,110 @@ mkdir -p cache || die
+HEADERS += \$\$files(../../../src/apps/$project/*.h, true)
+HEADERS += \$\$files(../../../nebula/*.h, true)
 
 SOURCES += i18nstrings_p.cpp
-SOURCES += ../i18nstrings.cpp
-SOURCES += \$\$files(../../src/shared/*.cpp, true)
-SOURCES += \$\$files(../../src/apps/vpn/*.cpp, true)
-SOURCES += \$\$files(../../nebula/*.cpp, true)
+SOURCES += ../../i18nstrings.cpp
+SOURCES += \$\$files(../../../src/shared/*.cpp, true)
+SOURCES += \$\$files(../../../src/apps/$project/*.cpp, true)
+SOURCES += \$\$files(../../../nebula/*.cpp, true)
 
 TRANSLATIONS += translations.ts
 
-RESOURCES += \$\$files(../../src/shared/*.qrc, true)
-RESOURCES += \$\$files(../../src/apps/vpn/*.qrc, true)
-RESOURCES += \$\$files(../../nebula/*.qrc, true)
+RESOURCES += \$\$files(../../../src/shared/*.qrc, true)
+RESOURCES += \$\$files(../../../src/apps/$project/*.qrc, true)
+RESOURCES += \$\$files(../../../nebula/*.qrc, true)
 EOF
-print G "done"
+  print G "done"
 
-QT_HOST_BINS=$(qmake6 -query QT_HOST_BINS)
+  QT_HOST_BINS=$(qmake6 -query QT_HOST_BINS)
 
-print Y "Generating the main translation file... "
-${QT_HOST_BINS}/lupdate translations/generated/dummy_ts.pro -ts translations.ts || die
+  print Y "$project - Generating the main translation file... "
+  ${QT_HOST_BINS}/lupdate translations/generated/$project/dummy_ts.pro -ts translations.ts || die
 
-printn Y "Generating strings for addons... "
-python scripts/addon/generate_all.py
-mkdir -p addon_ts || die
-cp addons/generated/addons/*.ts addon_ts
-print G "done."
+  printn Y "$project - Generating strings for addons... "
+  python3 scripts/addon/generate_all.py
+  mkdir -p addon_ts || die
+  cp addons/generated/addons/*.ts addon_ts
+  print G "done."
+
+  for branch in $(git branch -r | grep origin/releases); do
+    git checkout $branch &>/dev/null || die
 
-for branch in $(git branch -r | grep origin/releases); do
-  git checkout $branch &>/dev/null || die
-
-  printn Y "Importing main strings from $branch..."
-  python cache/generate_strings.py -o translations/generated translations/strings.yaml || die
-  ${QT_HOST_BINS}/lupdate translations/generated/dummy_ts.pro -ts branch.ts || die
-  ${QT_HOST_BINS}/lconvert -i translations.ts branch.ts -o tmp.ts || die
-  mv tmp.ts translations.ts || die
-  rm branch.ts || die
-
-  if [ -f "scripts/addon/generate_all.py" ]; then
-    printn Y "Importing addon strings from $branch..."
-    python scripts/addon/generate_all.py
-    ts_files="addons/generated/addons/*.ts"
-    for f in $ts_files
-    do
-      ts_name=$(basename "$f")
-      if [ -f "addon_ts/${ts_name}" ]; then
-        printn Y "File ${ts_name} exists, updating with branch strings..."
-        ${QT_HOST_BINS}/lconvert -i "addon_ts/${ts_name}" "addons/generated/addons/${ts_name}" -o tmp.ts || die
-        mv tmp.ts "addon_ts/${ts_name}"
+    printn Y "Importing main strings from $branch..."
+    if [ -f translations/strings.yaml ]; then
+      if [ $project = "vpn" ]; then
```

Comment:
This part of the script iterates over release branch.

If `translations/strings.yaml` exists, it means we're not on `main`, but on a release branch. In that case, we only need to extract the strings for the VPN project, otherwise the string for VPN 2.14 will end up in all other projects (e.g. Relay).

The `main` branch will have a `strings.yaml` file for each project, e.g. `translations/generated/vpn/translations/strings.yaml`.

Basically, we have a lot of code to deal with the existence of release branches, and we should probably clean it up at some point.